src/main/java/ru/mtsbank/fintech/animal_repository/AnimalRepositoryImpl.java

```diff
@@ -212,6 +211,57 @@ public <T> T readFromFile(String fileName, Class<T> type) {
         }
     }
 
+    /**
+     * <b>saveAnimal</b>
+     * Записывает данные полученные из CreateAnimalService в базу
+     */
+    private void saveAnimal(List<Animal> animalList) {
+        Session session = HibernateUtil.getSessionFactory().openSession();
+        Transaction transaction = null;
+
+        try {
+            transaction = session.beginTransaction();
+
+            for (Animal animal : animalList) {
+                session.save(animal.getAnimalType());
+                session.save(animal.getBreed());
+                session.save(animal);
+            }
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            e.printStackTrace();
+        } finally {
+            session.close();
+        }
```

Comment:
try with resources бы подошло лучше:

```suggestion
    try (Session session = HibernateUtil.getSessionFactory().openSession()) {
        Transaction transaction = session.beginTransaction();

        for (Animal animal : animalList) {
            session.save(animal.getAnimalType());
            session.save(animal.getBreed());
            session.save(animal);
        }
        transaction.commit();
    } catch (Exception e) {
        e.printStackTrace();
    }
```